crawl-ref/source/mutation.cc

```diff
@@ -97,62 +97,54 @@ static const body_facet_def _body_facets[] =
     { EQ_BOOTS, MUT_TALONS, 3 }
 };
 
+enum mutation_resolution
+{
+    MUTR_REPLACE, // Remove the existing mutation and replace with the new one.
+    MUTR_COEXIST, // Allow both mutations to exist.
+    MUTR_MODIFY,  // Delete a level of the existing mutation.
+    MUTR_IGNORE,  // Discard the new mutation, leaving existing mutations unchanged.
+};
+
+struct mut_conflict {
+    mutation_type       muts[2];
+    mutation_resolution forced;
+    mutation_resolution unforced;
+};
+
 /**
- * Conflicting mutation pairs. Entries are symmetric (so if A conflicts
- * with B, B conflicts with A in the same way).
- *
- * The third value in each entry means:
- *   0: If the new mutation is forced, remove all levels of the old
- *      mutation. Either way, keep scanning for more conflicts and
- *      do what they say (accepting the mutation if there are no
- *      further conflicts).
- *
- *  -1: If the new mutation is forced, remove all levels of the old
- *      mutation and scan for more conflicts. If it is not forced,
- *      fail at giving the new mutation.
- *
- *   1: If the new mutation is temporary, just allow the conflict.
- *      Otherwise, trade off: delete one level of the old mutation,
- *      don't give the new mutation, and consider it a success. 
- *
- * It makes sense to have two entries for the same pair, one with value 0
- * and one with 1: that would replace all levels of the old mutation if
- * forced, or a single level if not forced. However, the 0 entry must
- * precede the 1 entry; so if you re-order this list, keep all the 0s
- * before all the 1s.
+ * Conflicting mutation pairs and how to resolve them based on the nature of the incoming
+ * mutation (forced or unforced). Entries are symmetric (so if A conflicts with B, B
+ * conflicts with A in the same way).
```

Comment:
Worth commenting that the order is important, since that's the order in which conflicts are evaluated. (E.g., trying to add MUT_SLOW_HEALING while you already have both MUT_REGENERATION and MUT_NO_DEVICE_HEALING will have different effects depending on what order those two are listed in.)